effectful/ops/syntax.py

```diff
@@ -924,8 +908,25 @@ def __call__(self, *args: Expr, **kwargs: Expr) -> Expr[T]:
         return call(self, *args, **kwargs)  # type: ignore
 
 
-@defterm.register(collections.abc.Callable)
```

Comment:
Do we still need a `defterm` pattern for `Callable`? We don't want to trace the internals of a function by default, but we do want function application to go through `call` so we can add partial evaluation rules.

Reply:
No, I don't think so. That behavior comes from `CallableTerm`, which is introduced by `defdata`.

Reply:
Right, but doesn't addressing #272 require introducing `call` on all function applications? I was expecting something like this:
```py
@defterm.register(Callable)
def defterm_callable(v: Callable[P, T]) -> Callable[P, T]:
    return functools.partial(call, v)
```

Reply:
Does it? I think that would be new behavior. You should be able to get the effect of handling `call` by handling apply instead.